File: DB-Manager/sdk/db_manager.py
import requests


class DBManager:
    def __init__(self, host: str="localhost", port: int=3001) -> None:
        self.uri = f"http://{host}:{port}"
        self.endpoints = {
            "update_record": f"{self.uri}/update_record"
        }

    def update_record(self, hash_of_state: int, action: list[int]):

        res = requests.post(
            self.endpoints["update_record"],
            data = {
                "hash_of_state": hash_of_state,
                "action": action
            }
        )
        return res

File: DB-Manager/sdk/db_manager.js
// const fetch = require('node-fetch'); // if you working on Node but not Browser

class DBManager {
    constructor(host = "localhost", port = 3001) {
        this.uri = `http://${host}:${port}`;
        this.endpoints = {
            get_choice: `${this.uri}/get_choice`
        };
    }

    async get_choice(state) {
        
        const params = {
            state: state
        };

        const queryString = new URLSearchParams(params).toString();
        const res = await fetch(
            `${this.endpoints.get_choice}?${queryString}`,
            {mode: 'no-cors'}
        );
        const data = await res.json();
        return data;
    }
}

// Demo usage
const dm = new DBManager();

dm.get_choice("...........................OX......XO...........................")
    .then((res) => console.log(res))
    .catch((err) => console.error(err));